### DISCUSSION TITLE
How to manage two GitHub user on my local com?

### DISCUSSION THREAD

**Discussion by GreenCat1996** [community/community#69213] [New to GitHub]
### Select Topic Area

Question

### Body

How to Manage two GitHub accounts on your computer can be done by configuring Git to use different credentials for each account?

**Comment by mdavic0** [comment]
**Managing Multiple GitHub Accounts on Your Computer**

You can manage two GitHub accounts on your computer by configuring Git to use different credentials for each account. Follow these steps:

1. **Configure Git Global Settings**:
   - Open a terminal or command prompt.
   - Configure your global Git username and email for your first GitHub account:
     ```shell
     git config --global user.name "Your First Username"
     git config --global user.email "your.email@example.com"
     ```
   - Configure your global Git username and email for your second GitHub account using the same commands with the new credentials.

2. **Generate SSH Keys**:
   - Create separate SSH keys for each GitHub account to ensure proper authentication. Use the following commands to generate SSH keys:
     ```shell
     ssh-keygen -t rsa -C "your.email@example.com"
     ```
     - Save each key with a unique name and do not overwrite existing keys.

3. **Add SSH Keys to Your GitHub Accounts**:
   - Go to your GitHub account settings.
   - Under "SSH and GPG keys," add the public SSH key associated with the first GitHub account.
   - Do the same for the second GitHub account, adding the second SSH key.

4. **Configure SSH Identity**:
   - Create a configuration file in your home directory (e.g., `~/.ssh/config`) if it doesn't exist and add the following configuration:
     ```shell
     # Account 1
     Host github.com
     HostName github.com
     User git
     IdentityFile ~/.ssh/your_first_ssh_key

     # Account 2
     Host github-second
     HostName github.com
     User git
     IdentityFile ~/.ssh/your_second_ssh_key
     ```
   - Replace `your_first_ssh_key` and `your_second_ssh_key` with the actual paths to your SSH keys.

5. **Clone and Configure Repositories**:
   - When cloning repositories, use the appropriate host alias from your SSH config. For example:
     ```shell
     git clone git@github.com:username/repo.git # First account
     git clone git@github-second:username/repo.git # Second account
     ```

By following these steps, you can manage two GitHub accounts on your computer with different credentials and SSH keys. Be sure to use the correct repository URLs and SSH host aliases when working with each account's repositories.

*Note: Always replace placeholders like "Your First Username," "your.email@example.com," and file paths with your actual information.*
